zap/src/test/java/org/parosproxy/paros/network/HttpRequestHeaderUnitTest.java

```diff
@@ -240,6 +240,48 @@ void isCssShouldReturnTrueWhenUrlIndicatesCss(String url) {
         assertTrue(reqHeader.isCss());
     }
 
+    @ParameterizedTest
+    @ValueSource(
+            strings = {
+                "http://example.org/images/file.ext", // In directory path
+                "http://xrayimage.com/", // In domain name
+                "https://example.image", // In domain extension (TLD)
+                "https://example.image/dir/file.ext", // In domain extension (TLD)
+                "https://example.gif/dir/file.ext", // In domain extension (TLD)
+                "https://example.org/dir/file?foo=bar&thing=image", // In parameter value
+                "https://example.org/dir/file?foo=bar&thing=gif", // In parameter value
+                "http://example.org/image/file.ext?foo=bar&type=.image", // In parameter value
+                // including period
+                "http://example.org/image/file.ext?foo=bar&type=.gif", // In parameter value
+                // including period
+                "http://example.org/image/file.ext?foo=bar&thing=image.gif", // In parameter value,
+                // plausible filename
+                "https://example.org/dir/file?foo=bar&image=file.ext", // In parameter name
+                "https://example.org/dir/file?foo=bar&image=file.gif" // In parameter name and value
```

Comment:
Since the data is the same for CSS it would be better if these were extracted into a method (that allows to specify the extension) to avoid duplicating them (same for following test).

Reply:
I dont disagree but it isn't just the extensions, some are paths etc. Also I believe I came up with one or two additional test values for isImage

Reply:
I noticed but "image" is not relevant (at least for the code, fine if you want to test the name itself, can be configured in that case).

Reply:
I'd mainly included that because the repose isImage uses that as it's pattern